### ISSUE TITLE
Stricter types for values in `@directus/memory`

### ISSUE THREAD

**Issue by paescuj** [directus/directus#20893]
### Describe the Improvement

Currently, TypeScript allows any value (`T = unknown`) to be used within the `KV` store (thus also Cache and Bus) while the value actually needs to be de-/serializable.

https://github.com/directus/directus/blob/60ea729957bcd6de28c466fe4369346e14922073/packages/memory/src/kv/lib/local.ts#L19-L32

It would be good to restrict the types, otherwise we might run into runtime issues.

**Comment by br41nslug** [comment]
> It would be good to restrict the types, otherwise we might run into runtime issues.

Since we're serializing/deserializing data here I dont think we can solve that with TS types which only check at build time while deserializing will always be u`nknown` until runtime where the actual deserialization of data happens 🤔 this same issue happens when using `JSON.parse` which requires Zod/Joi/etc validation instead of typing

**Comment by rijkvanzanten** [comment]
I'm afraid @br41nslug is right here.. The only alternative would be for us to figure out some sort of way to make our own `"unknown"` that means "any JS primitive", but I'm not sure if that'll make the situation better or worse, as TS assigns special meanings to unknown/any

**Comment by rijkvanzanten** [comment]
I do prefer `unknown` over `any` though (like `JSON.parse()` has as the default). Explicitly saying "We don't know what the value is" feels better in this case then "It's whatever, ignore it"

**Comment by paescuj** [comment]
True, true, [JsonValue](https://github.com/directus/directus/blob/60ea729957bcd6de28c466fe4369346e14922073/packages/types/src/misc.ts#L32) is probably not really applicable here.

Could/should we wrap https://github.com/directus/directus/blob/60ea729957bcd6de28c466fe4369346e14922073/packages/memory/src/utils/serialize.ts#L25 in a try...catch statement then, maybe returning the string if not deserializable?

> I do prefer unknown over any though (like JSON.parse() has as the default). Explicitly saying "We don't know what the value is" feels better in this case then "It's whatever, ignore it"

Absolutely 👍
